temporal-sdk/src/main/java/io/temporal/internal/client/RootWorkflowClientInvoker.java

```diff
@@ -200,6 +170,151 @@ public WorkflowSignalWithStartOutput signalWithStart(WorkflowSignalWithStartInpu
     return new WorkflowSignalWithStartOutput(new WorkflowStartOutput(execution));
   }
 
+  @Override
+  public <R> WorkflowUpdateWithStartOutput<R> updateWithStart(
+      WorkflowUpdateWithStartInput<R> input) {
+
+    WorkflowStartInput startInput = input.getWorkflowStartInput();
+    DataConverter dataConverterWithWorkflowContext =
+        clientOptions
+            .getDataConverter()
+            .withContext(
+                new WorkflowSerializationContext(
+                    clientOptions.getNamespace(), startInput.getWorkflowId()));
+    StartWorkflowExecutionRequest startRequest =
+        toStartRequest(dataConverterWithWorkflowContext, startInput).build();
+
+    UpdateWithStartWorkflowOperation<R> updateOperation = input.getUpdateOperation();
+    UpdateOptions<?> updateOptions = updateOperation.getOptions();
+    updateOptions.validate();
+    StartUpdateInput<?> updateInput =
+        new StartUpdateInput<>(
+            WorkflowExecution.newBuilder().setWorkflowId(startInput.getWorkflowId()).build(),
+            Optional.of(startInput.getWorkflowType()),
+            updateOptions.getUpdateName(),
+            Header.empty(),
+            updateOptions.getUpdateId(),
+            updateOperation.getUpdateArgs(),
+            updateOptions.getResultClass(),
+            updateOptions.getResultType(),
+            "", // firstExecutionRunId is always empty
+            WaitPolicy.newBuilder()
+                .setLifecycleStage(updateOptions.getWaitForStage().getProto())
+                .build());
+    UpdateWorkflowExecutionRequest updateRequest =
+        toUpdateWorkflowExecutionRequest(updateInput, dataConverterWithWorkflowContext);
+
+    ExecuteMultiOperationRequest request =
+        ExecuteMultiOperationRequest.newBuilder()
+            .setNamespace(clientOptions.getNamespace())
+            .addOperations(
+                0,
+                ExecuteMultiOperationRequest.Operation.newBuilder()
+                    .setStartWorkflow(startRequest)
+                    .build())
+            .addOperations(
+                1,
+                ExecuteMultiOperationRequest.Operation.newBuilder()
+                    .setUpdateWorkflow(updateRequest)
+                    .build())
+            .build();
+
+    ExecuteMultiOperationResponse response;
+    try {
+      response = genericClient.executeMultiOperation(request);
```

Comment:
Do we not need to call this in a loop like regular update to avoid loosing the update requests ?

Reply:
Good callout. We do. I don't think I did that in Go (https://github.com/temporalio/sdk-go/blob/592ae12141a56d9c52a1cb40eb28fa06d5550254/internal/internal_workflow_client.go#L1718). I'll file a ticket for that.

Reply:
Okay, addressed here: https://github.com/temporalio/sdk-java/pull/2199/commits/3652e31b2a667047f8a6ea4b2c20a432e8acc6ea - I couldn't see a good way to test that, though?

Reply:
I think it is something that needs to be tested with a unit test mocking the grpc stub

Reply:
Is there a good example for this? I couldn't really find one yet.

Reply:
Addressed in https://github.com/temporalio/sdk-java/pull/2199/commits/67bfbb4916ed87585dcaa23596128f5f49fa4eb5